File: lib/api/rest/client/js/models/template.ts
export default class Template {
    public static fromRaw(data: any) {
        data ||= {}
        return new Template(
            data['name'] || '',
            data['data'] || '',
            data['meta'] || null,
            data['sub'] || null
        )
    }

    constructor(
        public name: string,
        public data: string,
        public meta: any,
        public sub: any
    ) {}
}

File: src/template/template.go
package template

import (
	"io"
	"yubin/common/data/kv"
)

type Template interface {
	SubTemplate(name string) (Template, bool)
	SubTemplates() map[string]Template
	WriteTo(writer io.Writer, data any) error
	Meta() map[string]any
	Raw() string
	Name() string
}

type Repo kv.Storage[Template]
